src/client/Client.cc

```diff
@@ -11080,6 +11049,23 @@ void Client::ll_register_callbacks(struct ceph_client_callback_args *args)
     umask_cb = args->umask_cb;
 }
 
+// This is deprecated, use ll_register_callbacks2() instead.
+void Client::ll_register_callbacks(struct ceph_client_callback_args *args)
+{
+  ceph_assert(!is_mounting() && !is_mounted() && !is_unmounting());
+
+  _ll_register_callbacks(args);
+}
+
+int Client::ll_register_callbacks2(struct ceph_client_callback_args *args)
```

Comment:
Do we really need a ll_register_callbacks2? Could we just change the prototype of ll_register_callbacks() to return int instead? The existing callers should just ignore the return code anyway. They may need a build-time test to see what the thing returns, but they'll need one to see whether ll_register_callbacks2 exists anyway.

Reply:
> Do we really need a ll_register_callbacks2? Could we just change the prototype of ll_register_callbacks() to return int instead? The existing callers should just ignore the return code anyway. They may need a build-time test to see what the thing returns, but they'll need one to see whether ll_register_callbacks2 exists anyway.

I was thinking whether there has any other third part APP is also using the libcephfs. And I am okay to remove the `ll_register_callbacks2`.